src/st/no0/e_skelerang.c

```diff
@@ -2,17 +2,351 @@
+                if (self->facingLeft) {
+                    entity->posX.i.hi -= D_us_80181EA4[index].x;
+                } else {
+                    entity->posX.i.hi += D_us_80181EA4[index].x;
+                }
+                entity->posY.i.hi += D_us_80181EA4[index].y;
+            }
+            DestroyEntity(self);
+        }
+        break;
+    case SKELERANG_DEBUG:
+#include "../pad2_anim_debug.h"
+    }
+
+    if (self->animCurFrame > 39) {
+        self->hitboxOffX = -1;
+        self->hitboxOffY = 16;
+        self->hitboxWidth = 8;
+        self->hitboxHeight = 8;
+    } else {
+        self->hitboxOffX = 0;
+        self->hitboxOffY = 2;
+        self->hitboxWidth = 5;
+        self->hitboxHeight = 21;
+    }
+}
+
+void EntitySkelerangBoomerang(Entity* self) {
+    Entity* entity;
+    u8 step;
+    s16 angle;
+    s16 posX;
+    s16 posY;
+
+    switch (self->step) {
+    case BOOMERANG_INIT:
+        InitializeEntity(g_EInitSkelerangBoomerang);
+        self->drawFlags |= FLAG_DRAW_ROTZ;
+        self->animCurFrame = 0;
+        self->hitboxState = 0;
+        break;
+    case BOOMERANG_PRETHROW:
+        entity = (self - 2) - self->params;
+        self->posX.i.hi = entity->posX.i.hi;
+        self->posY.i.hi = entity->posY.i.hi;
+        self->hitboxState = 0;
+        self->animCurFrame = 0;
+        self->step_s = 0;
+        self->ext.skelerang.unk84 = 48;
+        if (self->params) {
+            self->ext.skelerang.angle = 0;
+            return;
+        }
+        self->ext.skelerang.angle = 128;
+        break;
+    case BOOMERANG_FLY:
+        self->hitboxState = 1;
+        MoveEntity();
+        self->rotZ += 256;
+        if (!self->ext.skelerang.unk84) {
+            self->step_s = 1;
+        } else {
+            self->ext.skelerang.unk84--;
+        }
+        entity = (self - 1) - self->params - self->step_s;
+        angle = GetAngleBetweenEntitiesShifted(self, entity);
+        step = self->step_s + 3;
+        self->ext.skelerang.angle =
+            AdjustValueWithinThreshold(step, self->ext.skelerang.angle, angle);
+        SetEntityVelocityFromAngle(self->ext.skelerang.angle, 48);
+        posX = entity->posX.i.hi - self->posX.i.hi;
+        posY = entity->posY.i.hi - self->posY.i.hi;
+        posX = abs(posX);
+        posY = abs(posY);
+        if (posX < 6 && posY < 6) {
+            PlaySfxPositional(SFX_MULTI_CLOCK_TICK);
+            self->step_s++;
+        }
+        if (self->step_s == 2) {
+            self->step++;
+        }
+        break;
+    case BOOMERANG_IN_HAND:
+        self->hitboxState = 0;
+        self->rotZ = 512;
+        break;
+    case BOOMERANG_DESTROY:
+        entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (entity != NULL) {
+            CreateEntityFromEntity(2, self, entity);
+            entity->params = 1;
+        }
+        DestroyEntity(self);
+        break;
+    }
+}
+
+void EntitySkelerangUnknown(Entity* entity) {
```

Comment:
Even in debugger I wasn't really able to ascertain what this is. It gets initialised as you approach the skelerangs but other than that I can't tell what it's doing. I can't trigger the destroy function in gameplay so this is a bit of a mystery